### PR TITLE
mvp features in cucumber form

### PR DESCRIPTION
**Pull request by wenzowski**
The list of MVP-targeted features can be reviewed with a `yarn cucumber:pretty` and are all tagged with an `@mvp` if scheduled for inclusion.

This preliminary list is based on the 70/30 MVP diagram reviewed yesterday:
Image-1

### CODE REVIEW

**Comment by wenzowski** [comment]
@rgreensp we probably need you on this repo for review

**Comment by hamzajaved** [comment]
Design file for User Flows
[CIIP User Flow.zip](https://github.com/bcgov/cas-ciip-portal/files/3837944/CIIP.User.Flow.zip)

**Comment by wenzowski** [comment]
@tenlig want to take a look?

File: app/features/analyst/ciip-filings.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
+@mvp
+Feature: Authentication
+
+  As an incentive analyst
+  I want triage the received CIIP filings list
```
Comment (tenlig) [inline app/features/analyst/ciip-filings.feature]: I want 'to'

File: app/features/analyst/ciip-filings.feature [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  So that I can analyze filed CIIP applications
+
+  Scenario: view dashboard of all received CIIP filings
+  Scenario: filter dashboard of all received CIIP filings by year
+  Scenario: filter dashboard of all received CIIP filings by status
+  Scenario: view details of a received CIIP filing
```
Comment (tenlig) [inline app/features/analyst/ciip-filings.feature]: Does viewing the details of a received CIIP filing include exposing the changes made by the industry reporter to SWRS imported data?  If so, should we be more explicit?  If not, can we add or is this not what we want anymore?

File: app/features/industry/authentication.feature [hunk 3]
```diff
@@ -0,0 +1,14 @@
+@mvp
+Feature: Authentication
+
+  As an industry user
```
Comment (tenlig) [inline app/features/industry/authentication.feature]: industry 'reporter'

File: app/features/industry/dry.feature [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  As an industry user
+  I want data I have already reported be re-used wherever possible
+  So that I do not have to repeat myself
+
+  Scenario: industry user is prompted to import data from a specific SWRS Emissions report (by date)
+  Scenario: admin data from SWRS is re-imported for every filing
+  Scenario: emission data from SWRS is re-imported for every filing
+  Scenario: fuel data from SWRS is re-imported for every filing
+  Scenario: electricity and heat data from SWRS is re-imported for every filing
+  Scenario: production data from SWRS is re-imported for every filing
```
Comment (tenlig) [inline app/features/industry/dry.feature]: I have the understanding that this info is from CIIP only?

File: app/features/industry/ciip-filing.feature [hunk 5]
```diff
@@ -0,0 +1,16 @@
...
+
+  Scenario: know when the CIIP application window will open
+  Scenario: start a CIIP application for a facility that hasn't filed yet
+  Scenario: automatically lock a CIIP application so others can't update it while I am updating it
+  Scenario: resume a CIIP application for a facility that has a filing in progress
+  Scenario: confirm all data reported in my CIIP application
+  Scenario: request certification of data reported in CIIP application
+  Scenario: correct fields an incentive analyst has flagged for review
+  Scenario: receive confirmation that my CIIP application has been filed
+  Scenario: review filed CIIP applications
```
Comment (tenlig) [inline app/features/industry/ciip-filing.feature]: Does this include viewing the status of their filed CIIP applications? If so then we need to mark GGIRCS-571 as a duplicate of this scenario in youtrack

File: app/features/executive/availability.feature [hunk 6]
```diff
@@ -0,0 +1,14 @@
+Feature: Availability
+
+  As an executive
+  I want the CIIP portal to be available at all times
+  So that the service design is designed effectively
```
Comment (tenlig) [inline app/features/executive/availability.feature]: I'm not sure I quite understand the explicitness behind this reasoning?

File: app/features/ciip-mvp.md [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+  - ability for reporter to view flagged data and [correct it](./industry/ciip-filing.feature)
+  - can receive [email notifications](./industry/email.feature) from CIIP portal
+  - email notification could contain a link to [flagged data for correction](./industry/email.feature)
+  - [certification](./industry/ciip-filing.feature) request
+  - [legal disclaimer](./industry/registration.feature)
+  - incorporation of feedback from first user study [PENDING]
+
+System Admin
+------------
+
```
Comment (tenlig) [inline app/features/ciip-mvp.md]: Admin registration and authentication both need to be linked to this section

File: app/features/analyst/ciip-filings.feature [hunk 8]
```diff
@@ -0,0 +1,16 @@
+@mvp
+Feature: Authentication
```
[the remark this answers is not in the document]
Comment (wenzowski) [reply app/features/analyst/ciip-filings.feature]: fixed

File: app/features/admin/registration.feature [hunk 9]
```diff
@@ -0,0 +1,12 @@
+@mvp
+Feature: Registration
+
+  As system admin
```
[the remark this answers is not in the document]
Comment (wenzowski) [reply app/features/admin/registration.feature]: fixed

File: app/features/admin/authentication.feature [hunk 10]
```diff
@@ -0,0 +1,9 @@
+@mvp
+Feature: Authentication
+
+  As system admin
```
[the remark this answers is not in the document]
Comment (wenzowski) [reply app/features/admin/authentication.feature]: fixed

File: app/features/admin/products.feature [hunk 11]
```diff
@@ -0,0 +1,11 @@
+@mvp
+Feature: Products
+
+  As system admin
+  I want to be able to modify critical settings without redeploying the application
```
[the remark this answers is not in the document]
Comment (wenzowski) [reply app/features/admin/products.feature]: fixed

File: app/features/analyst/ciip-filings.feature (hunk 1 above)
Comment (wenzowski) [reply app/features/analyst/ciip-filings.feature]: fixed

File: app/features/industry/authentication.feature (hunk 3 above)
Comment (wenzowski) [reply app/features/industry/authentication.feature]: fixed

File: app/features/industry/dry.feature (hunk 4 above)
Comment (wenzowski) [reply app/features/industry/dry.feature]: fixed

File: app/features/analyst/ciip-filings.feature (hunk 2 above)
Comment (wenzowski) [reply app/features/analyst/ciip-filings.feature]: fixed

Comment (wenzowski) [reply app/features/analyst/ciip-filings.feature]: Is that diff of SWRS<=>CIIP data in scope for our MVP though? @Maralsotoudehnia @hamzajaved

File: app/features/industry/ciip-filing.feature (hunk 5 above)
Comment (wenzowski) [reply app/features/industry/ciip-filing.feature]: fixed

File: app/features/executive/availability.feature (hunk 6 above)
Comment (wenzowski) [reply app/features/executive/availability.feature]: the intent here is that it's difficult to deliver an effective service if the service is offline; updated

File: app/features/ciip-mvp.md (hunk 7 above)
Comment (wenzowski) [reply app/features/ciip-mvp.md]: This is a transitional document, aimed only at making it easier to reconcile these features against our poster...which didn't include admin registration and authentication. Are those now in scope for our MVP? @Maralsotoudehnia @hamzajaved

File: app/features/analyst/ciip-filings.feature (hunk 2 above)
Comment (tenlig) [reply app/features/analyst/ciip-filings.feature]: We clarified this scope in our MVP discussion and wrote it on the paper as "expose changes to SWRS made in CIIP application (for analyst, not compliance)"

File: app/features/ciip-mvp.md (hunk 7 above)
Comment (tenlig) [reply app/features/ciip-mvp.md]: Once answered, if they are NOT mvp then you should remove your @mvp tag on both of them :)

Comment (wenzowski) [reply app/features/ciip-mvp.md]: ok, so this needs @Maralsotoudehnia clarification to address

Comment (hamzajaved) [reply app/features/ciip-mvp.md]: Worth a quick discussion in person with @Maralsotoudehnia

**Comment by wenzowski** [comment]
It's a very good question, and I've included it here because it's in the wireframes. If we can get a discussion with @Maralsotoudehnia sooner rather than later it would be a good idea since we've scheduled stories for products into this sprint and @dleard is already working on them

**Comment by wenzowski** [comment]
@hamzajaved the wireframes you shared here were too big to check in to this repo so I've put them on SharePoint in the `/Project Work/Wireframes` folder and extracted each artboard as a png for those who do not have XD into `/Project Work/Wireframes/export`.

File: app/features/ciip-mvp.md (hunk 7 above)
Comment (Maralsotoudehnia) [reply app/features/ciip-mvp.md]: I thought admin registration and authentication (i.e., security things) were going to be ongoing efforts that can only be fully implemented after MVP. @wenzowski am I misinterpreting? I can jump on a call before 3 today. Let me know if you want to discuss.

File: app/features/executive/business-continuity.feature [hunk 12]
```diff
@@ -0,0 +1,13 @@
+@mvp
```
Comment (Maralsotoudehnia) [inline app/features/executive/business-continuity.feature]: Stretch goal for MVP -- let's try to get this done but recognize that we may have to punt some scenarios.

File: app/features/industry/registration.feature [hunk 13]
```diff
@@ -0,0 +1,12 @@
...
+
+  As an industry reporter
+  I want to be able to onboard myself to the CIIP application portal
+  So that I can access business confidential information
+
+  Scenario: invalid session redirects to login page
+  Scenario: import user data from valid session after login
+  Scenario: re-import user data from valid session after login when session attributes have changed
+  Scenario: registration page allows the user to edit their details
+  Scenario: accept the legal disclaimer
```
Comment (Maralsotoudehnia) [inline app/features/industry/registration.feature]: @tenlig please make sure we schedule the feature for MVP -- talked to @wenzowski  and it is in scope.

File: app/features/admin/authentication.feature [hunk 14]
```diff
@@ -0,0 +1,9 @@
+@mvp
```
Comment (tenlig) [inline app/features/admin/authentication.feature]: @Maralsotoudehnia  is saying that this feature is not MVP

Comment (wenzowski) [reply app/features/admin/authentication.feature]: ...really? @Maralsotoudehnia what makes authentication as an administrator different from authentication as an analyst such that the admin story is out of scope but the analyst story is in scope?

**Pull request merged**